GEOSldas_App/util/postproc/ObsFcstAna_stats/Save_monthlysums.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+
+"""
+Sample script for computing and storing monthly sums and sums of squares and cross-products
+from GEOSldas ObsFcstAna output.  Data assimilation diagnostics such as the mean and
+std-dev of the observation-minus-forecast residuals can be diagnosed quickly from these
+intermediate "sums" files. 
+
+Usage:
+    First, edit "user_config.py" with proper experiments information.
+    Then, run this script as follows (on Discover):
+    $ module load python/GEOSpyD
+    $ ./Save_monthlysums.py
+    
+    # Background execution:
+    $ nohup ./Save_monthlysums.py > out.log &
+
+Authors: Q. Liu, R. Reichle, A. Fox
+Last Modified: May 2025
+"""
+
+import sys;       sys.path.append('../../shared/python/')
+import warnings;  warnings.filterwarnings("ignore")
+import os
+
+import numpy as np
+
+from datetime               import timedelta
+from postproc_ObsFcstAna    import postproc_ObsFcstAna
+from user_config            import get_config              # user-defined config info
+
+# ---
+#
+# If the script is run in the background, uncomment the following lines to see the redirected
+#   standard output in the out.log file immediately.  When the lines are commented out, the redirected
+#   standard output will not appear in the out.log file until the job has completed.
+# If the script is run in the foreground, the lines must be commented out.
+#
+#import io
+#sys.stdout = io.TextIOWrapper(open(sys.stdout.fileno(), 'wb', 0), write_through=True)
+#sys.stderr = io.TextIOWrapper(open(sys.stderr.fileno(), 'wb', 0), write_through=True)
+#
+# ---
+
+def main():
+     
+    config     = get_config()                 # edit user-defined inputs in user_config.py
+    
+    exp_list   = config['exp_list']
+    start_time = config['start_time']
+    end_time   = config['end_time']
+    obs_from   = config['obs_from']
+    sum_path   = config['sum_path']
+    out_path   = config['out_path']
+     
+    # ------------------------------------------------------------------------------------
+    # Postprocess raw ObsFcstAna output data into monthly sums 
+
+    # Initialize the postprocessing object
+    postproc = postproc_ObsFcstAna(exp_list, start_time, end_time, obs_from=obs_from, sum_path=sum_path)
+
+    # Compute and save monthly sums
+    postproc.save_monthly_sums()
+
+    # --------------------------------------------------------------------------------------
+    # Optionally compute long-term temporal/spatial statistics and create plots.
+    # The plotting scripts can also run standalone using the individual Plot_stats_*.py scripts,
```

Comment:
I like the optional plotting below.  I don't think we need to include main() functions in the Plot_stats*.py files.  Perhaps the best way to organize the code is to create a single "plot_OmF_stats.py" that contains the functions defined in the current Plot_stats*.py scripts.  Perhaps we can add a switch that lets the user skip the "save_monthly_sums()" action, in case all they want to do is plot.  Does that make sense?